src/libraries/System.Net.Ping/src/System/Net/NetworkInformation/Ping.RawSocket.cs

```diff
@@ -239,6 +243,10 @@ private async Task<PingReply> SendIcmpEchoRequestOverRawSocketAsync(IPAddress ad
                 catch (SocketException ex) when (ex.SocketErrorCode == SocketError.TimedOut)
                 {
                 }
+                catch (SocketException ex) when (ex.SocketErrorCode == SocketError.MessageSize)
+                {
+                    return new PingReply(new IPAddress(0), null, IPStatus.PacketTooBig, 0, Array.Empty<byte>());
```

Comment:
Maybe we should modify the CreateTimedOutPingReply() helper to instead be CreatePingReply(IPStatus status) ?

Reply:
Yep, I was planning to do that before moving this out of draft.